### PR TITLE
GH-37155: [MATLAB] Use `arrow.internal.validate.index.numeric()` in the `column()` method of `arrow.tabular.RecordBatch`

### LINKED ISSUE #37155: MATLAB: Use `arrow.internal.validate.index.numeric()` in the `column()` method of `arrow.tabular.RecordBatch`
### Describe the enhancement requested

Now that the PR introducing the `arrow.internal.validate.inex.numeric()` function has been merged (#37150), we should use this utility function within the `column()` method of `arrow.tabular.RecordBatch`. Doing so will remove duplicate code from the code base.

### Component(s)

MATLAB

### PR DESCRIPTION
**Pull request by sgilmore10**
### Rationale for this change

Now that the PR introducing the `arrow.internal.validate.index.numeric()` function has been merged (#37150), we should use this utility function within the `column()` method of `arrow.tabular.RecordBatch`. Doing so will remove duplicate code from the code base.

### What changes are included in this PR?

1. Updated the `column()` method of `arrow.tabular.RecordBatch` to use `arrow.internal.validate.index.numeric()`. 

### Are these changes tested?

Yes, these changes are covered by these two existing test classes:

1. `test/arrow/tabular/tRecordBatch.m`
2. `test/arrow/internal/validate/index/tNumeric.m`

In addition, I added a new test point to `tRecordBatch.m` called `ErrorIfNonPositiveIndex` to verify `arrow.internal.validate.index.numeric` is integrated correctly.

### Are there any user-facing changes?

No.

### Future Directions

1. In a future pull request, we will add support for indexing columns by name. Currently, the `column()` method only supports numeric indexing.
* Closes: #37155

### CODE REVIEW

File: matlab/test/arrow/tabular/tRecordBatch.m [hunk 1]
```diff
@@ -119,7 +119,21 @@ function UnsupportedColumnIndexType(tc)
...
+        end
+
+        function ErrorIfIndexIsNonScalar(tc)
+            TOriginal = table(1, 2, 3);
+            arrowRecordBatch = arrow.recordbatch(TOriginal);
+            fcn = @() arrowRecordBatch.column([1 2]);
+            tc.verifyError(fcn, "MATLAB:expectedScalar");
+        end
+
+        function ErrorIfNonPositiveIndex(tc)
```
Comment (kevingurney) [inline matlab/test/arrow/tabular/tRecordBatch.m:132]: To be consistent with the naming of `ErrorIfIndexIsNonScalar`, perhaps we should rename this to `ErrorIfIndexIsNonPositive`?

**Review by kevingurney — CHANGES_REQUESTED** [review]
This is a nice improvement - thank you!

File: matlab/test/arrow/tabular/tRecordBatch.m (hunk 1 above)
Comment (sgilmore10) [reply matlab/test/arrow/tabular/tRecordBatch.m]: Good point.

**Review by kevingurney — APPROVED** [review]

**Comment by kevingurney** [comment]
+1

**Pull request merged**